ctapipe/core/qualityquery.py

```diff
@@ -132,11 +138,13 @@ def get_table_mask(self, table):
         mask : np.ndarray[bool]
             Boolean mask of valid entries.
         """
-        n_criteria = len(self.quality_criteria) + 1
+        n_criteria = len(self.criteria_names) + 1
         result = np.ones((n_criteria, len(table)), dtype=bool)
 
-        for i, res in enumerate(self.engine(table), start=1):
-            result[i] = res
+        for i, (_, engine) in enumerate(self.engines, start=1):
+            for tel_id in np.unique(table["tel_id"]):
```

Comment:
Is this faster than group_by and then resorting to the original order? It seems like the more costly operation...

Reply:
I didn't do it at first as astropy table group_by does not give you the corresponding tel_id, as pandas does it. I can check

Reply:
See https://docs.astropy.org/en/stable/table/operations.html#manipulating-groups

```
for key, group in zip(obs_by_name.groups.keys, obs_by_name.groups):
```

Reply:
changed it to use group_by and not rely on masks